### ISSUE TITLE
VUID-VkGraphicsPipelineCreateInfo-pStages-06896 is validated incorrectly

### ISSUE THREAD

**Issue by Rua** [KhronosGroup/Vulkan-ValidationLayers#5658]
This VUID says:
> If the pipeline is being created with pre-rasterization shader state, all elements of `pStages` **must** have a stage set to a shader stage which participates in fragment shader state or pre-rasterization shader state

In layman's terms, this is saying that all the provided shaders must be graphics shaders, they can't be e.g. compute or raytracing shaders. But the actual code is testing something else:
https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/ddd3f7ee9d6d87307e3e85e8860d085880998068/layers/core_checks/pipeline_cc_validation.cpp#L1527-L1532

This is testing whether at least one pre-raster shader has been provided, which is very different. This would not catch the user providing a compute shader, for example. Fragment shaders are also not included in the check.

**Comment by spencer-lunarg** [comment]
Before I write a test to see if this works, I have to believe there is another VU that would blow up first

Did you hit this VU in some code or are you just bringing up this from eye-ing the code (both is valid)

**Comment by Rua** [comment]
I got it from looking at the code itself.

**Comment by spencer-lunarg** [comment]
Tried to write a test, it is not possible to hit this, we are already guarding against things like a compute stage in the `VkGraphicsPipelineCreateInfo::pStages`

Also we have tests for this VU so seems the coverage is working as expected

**Comment by spencer-lunarg** [comment]
> Fragment shaders are also not included in the check.

this is a spec bug, getting it fixed now
